packages/react-query/src/__tests__/useQuery.test.tsx

```diff
@@ -5729,6 +5730,7 @@ describe('useQuery', () => {
           queryFn: async ({ signal }) => {
             count++
             await sleep(10)
+            // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
             return `${signal ? 'signal' : 'data'}${count}`
```

Comment:
here we could remove the `signal` because destructuring it from `ctx` "counts" as the signal being consumed:

```suggestion
            return `signal${count}`
```